include/communication/service/communication_service.h

```diff
@@ -0,0 +1,44 @@
+#ifndef COMMUNICATION_SERVICE_H
+#define COMMUNICATION_SERVICE_H
+
+#include "i_service.h"
+#include <QObject>
+
+#include <QScopedPointer>
+#include <QVector>
+
+#include "communication.h"
+#include "communication_description.h"
+#include "i_communication_protocol.h"
+#include "i_json_source.h"
+#include "protocol_specification.h"
+
+namespace md::app
+{
+class CommunicationService
+    : public QObject
+    , public md::domain::IService
+{
+    Q_OBJECT
+
+public:
+    CommunicationService(const QString& fileName);
+
+    void registerProtocol(const QString& name, data_source::ICommunicationProtocol* protocol);
```

Comment:
const QString& id

Reply:
At this time, protocol specification holds name, not ID, as Jerom. We should not change it in one place only

Reply:
In futer will be need both id and name